TeachingAssistantProfesor/app/src/main/java/malvolyo/teachingassistantprofesor/ActiveQuizActivity.java

```diff
@@ -0,0 +1,146 @@
+package malvolyo.teachingassistantprofesor;
+
+import android.content.Intent;
+import android.graphics.Color;
+import android.support.v7.app.AppCompatActivity;
+import android.os.Bundle;
+import android.view.View;
+
+import com.github.mikephil.charting.charts.PieChart;
+import com.github.mikephil.charting.components.Description;
+import com.github.mikephil.charting.data.PieData;
+import com.github.mikephil.charting.data.PieDataSet;
+import com.github.mikephil.charting.data.PieEntry;
+import com.github.mikephil.charting.formatter.PercentFormatter;
+import com.google.firebase.database.DataSnapshot;
+import com.google.firebase.database.DatabaseError;
+import com.google.firebase.database.DatabaseReference;
+import com.google.firebase.database.FirebaseDatabase;
+import com.google.firebase.database.ValueEventListener;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+
+public class ActiveQuizActivity extends AppCompatActivity {
+    private PieChart pc;
+    private DatabaseReference database;
+
+    private String user;
+    private String course;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_active_quiz);
+
+        Intent intent = getIntent();
+        user = intent.getStringExtra(Constants.USERNAME);
+        course = intent.getStringExtra(Constants.COURSE);
+
+        pc = (PieChart)findViewById(R.id.quizPieChart);
+
+        pc.setUsePercentValues(true);
+
+        pc.setDrawHoleEnabled(true);
+        pc.setHoleRadius(10);
+        pc.setTransparentCircleRadius(10);
+
+        Description desc = new Description();
+        desc.setText("Quiz Statistics");
+        desc.setTextColor(Color.BLACK);
+        desc.setTextSize(20);
+        pc.setDescription(desc);
+
+
+        database = FirebaseDatabase.getInstance().getReference();
+
+        database.child(course).child(DBConstants.QUIZ).child(DBConstants.ACTIVE_QUIZ).
+                child(DBConstants.ANSWERED).addValueEventListener(new ValueEventListener() {
+                    @Override
+                    public void onDataChange(DataSnapshot dataSnapshot) {
+                        int counter = 0;
+
+                        pc.setVisibility(View.GONE);
+                        findViewById(R.id.activeQuizLoading).setVisibility(View.VISIBLE);
```

Comment:
You should keep references to views as members in the Activity. findViewById() is a costly method to call every time. This goes for all of your views.